TeamCode/src/main/java/org/firstinspires/ftc/teamcode/controls/def/button/ButtonCtlDef.java

```diff
@@ -0,0 +1,49 @@
+package org.firstinspires.ftc.teamcode.controls.def.button;
+
+import com.arcrobotics.ftclib.gamepad.GamepadEx;
+import com.arcrobotics.ftclib.gamepad.GamepadKeys;
+
+import org.firstinspires.ftc.teamcode.controls.def.ControlDefinition;
+
+import java.util.function.Consumer;
+
+public class ButtonCtlDef implements ControlDefinition {
+    GamepadKeys.Button button;
+    ButtonCtlTrigger trigger;
+    boolean returnOnlyOnTrue;
+    Consumer<Boolean> action;
+
+    public ButtonCtlDef(GamepadKeys.Button button, ButtonCtlTrigger trigger, boolean returnOnlyOnTrue, Consumer<Boolean> action) {
+        this.button = button;
+        this.trigger = trigger;
+        this.returnOnlyOnTrue = returnOnlyOnTrue;
+        this.action = action;
+    }
+
+    public ButtonCtlDef(GamepadKeys.Button button, ButtonCtlTrigger trigger, Consumer<Boolean> action) {
+        this(button, trigger, false, action);
+    }
+
+    @Override
+    public void run(GamepadEx gamepad) {
+        boolean result;
+        switch (trigger) {
```

Comment:
It feels like using a [ButtonReader](https://docs.ftclib.org/ftclib/features/gamepad-extensions#buttonreader) would be more semantically clear of what is being done here

Reply:
but I would need to create a new ButtonReader class on every call of run(), idk if efficient

Reply:
I thought of maybe providing GamepadEx as a property of class rather than just passing it as an argument to `run`

Reply:
passing gamepadex to every control definition is stupid :sob: